File: July-LeetCoding-Challenge/20-Remove-Linked-List-Elements/Remove-Linked-List-Elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        prev = ListNode(-1, head)
        h = prev
        cur = head
        while(cur):
            while (cur and cur.val == val):
                cur = cur.next
            prev.next = cur
            prev = cur
            if (cur):
                cur = cur.next
        return h.next

File: July-LeetCoding-Challenge/20-Remove-Linked-List-Elements/Remove-Linked-List-Elements.cpp
class Solution1 {
public:
    ListNode* removeElements(ListNode* head, int val) {
        
        head = new ListNode(-1,head);
        ListNode* prev = head, *curr = head->next;
        
        while(curr){
            if(curr->val == val)
                prev->next = prev->next->next;
            else
                prev = curr;
            curr = curr->next;
        }
        
        return head->next;
    }
};
